File: python/ddj_a078.py

t = int(input())
for i in range(t):
	lst = []
	n = int(input())
	for j in range(n):
		a, b = input().split()
		a = int(a)
		b = int(b)
		if a > b:
			continue
		lst.append([b, a])
	lst.sort()
	cnt = 0
	time = 0
	f = [0]
	k = 0
	for j in range(lst[len(lst) - 1][0] + 1):
		if j > 0:
			f.append(f[j-1])
		while k < len(lst) and lst[k][0] <= j:
			f[j] = max(f[j], f[j - lst[k][1]] + 1)
			k += 1
	#print(f)
	print(f[lst[-1][0]])

	#dp
	#以期限sort，dp轉移式：
	#f(k) = max(f(k), f(k - time) + 1)

File: ddj_a078.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int t;
    cin >> t;
    
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
		pair<int, int> lst[n];
        
        for (int j = 0; j < n; j++) {
			cin >> lst[j].second >> lst[j].first;
		}
        
        sort(lst, lst + n);
        
        int cnt = 0;
        int time = 0;
       	int f[lst[n - 1].first + 1];
		f[0] = 0;
        int k = 0;
        
        for (int j = 1; j <= lst[n - 1].first; j++) {            
            while (k < n && lst[k].first <= j) {
                f[j] = max(f[j], f[j - lst[k].second] + 1);
                k++;
            }
        }
        
        cout << f[lst[n - 1].first] << endl;
    }
    
    return 0;
}

